roles/variables/defaults/main.yml

```diff
@@ -416,7 +437,115 @@ kafka_controller_ssl_enabled: "{{ssl_enabled}}"
 kafka_controller_ssl_mutual_auth_enabled: "{{ssl_mutual_auth_enabled}}"
 
 ### SASL Mechanism for controller Server to Server and Server to Client Authentication. Options are none, kerberos, digest. Server to server auth only working for digest-md5
-kafka_controller_sasl_protocol: "{{sasl_protocol if sasl_protocol == 'kerberos' else 'none'}}"
+kafka_controller_sasl_protocol: "{{sasl_protocol}}"
+
+# Uses custom filter to create a list of all sasl_protocols, removes ['none'], and reduces to unique items
+kafka_controller_sasl_enabled_mechanisms: "{{ kafka_controller_listeners | confluent.platform.get_sasl_mechanisms(kafka_controller_sasl_protocol) | difference(['none']) | unique }}"
+
+### Set this variable to customize the Linux User that the Kafka Broker Service runs with. Default user is cp-kafka.
+kafka_controller_user: "{{kafka_controller_default_user}}"
+
+### Set this variable to customize the Linux Group that the Kafka Broker Service user belongs to. Default group is confluent.
+kafka_controller_group: "{{kafka_controller_default_group}}"
+
+### Set this variable to customize the directory that the Kafka Broker writes log files to. Default location is /var/log/kafka.
+kafka_controller_log_dir: "{{kafka_controller_default_log_dir}}"
+
+kafka_controller_packages:
+  - confluent-common
+  - confluent-rest-utils
+  - confluent-metadata-service
+  - confluent-ce-kafka-http-server
+  - confluent-kafka-rest
```

Comment:
rest-utils metadata-service and kafka-rest might not be needed. We can try without those and confirm.

Reply:
yes, it works without those. Removed them